receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +66,364 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+	// with the row index of the SNMP data
+	resourceAttributeNames := configHelper.getResourceAttributeNames(data.parentOID)
+	resourceKey := getResourceKey(resourceAttributeNames, indexString)
+
+	// Create a new resource if needed
+	resource := metricHelper.getResource(resourceKey)
+	if resource == nil {
+		metricHelper.createResource(resourceKey, resourceAttributes)
+	}
+
+	// Create a new metric if needed
+	metric := metricHelper.getMetric(resourceKey, metricName)
+	if metric == nil {
+		if _, err := metricHelper.createMetric(resourceKey, metricName, metricCfg); err != nil {
+			return err
+		}
+	}
+
+	// Add data point to metric
+	if _, err := metricHelper.addMetricDataPoint(resourceKey, metricName, metricCfg, data, dataPointAttributes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getScalarDataPointAttributes returns the key value pairs of attributes for a given metric config scalar OID
+func getScalarDataPointAttributes(configHelper *configHelper, oid string) map[string]string {
+	dataPointAttributes := map[string]string{}
+	for _, attribute := range configHelper.metricAttributesByOID[oid] {
+		attributeKey := attribute.Name
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
+		dataPointAttributes[attributeKey] = attribute.Value
+	}
+
+	return dataPointAttributes
+}
+
+// getDataPointAttributes gets attributes for this metric's datapoint based on the previously
+// gathered attributes.
+// Keys will be determined from the related attribute config and values will come a few
+// different places.
+// Enum attribute value - comes from the metric config's attribute data
+// Indexed prefix attribute value - comes from the current SNMP data's index and the attribute
+// config's prefix value
+// Indexed OID attribute value - comes from the previously collected indexed attribute data
+// using the current index and attribute config to access the correct value
+func getIndexedDataPointAttributes(
+	configHelper *configHelper,
+	parentOID string,
+	indexString string,
+	indexedAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	datapointAttributes := map[string]string{}
+
+	for _, attribute := range configHelper.getMetricConfigAttributes(parentOID) {
+		attributeKey := attribute.Name
+		attributeValue := attribute.Value
+		if prefix := configHelper.getAttributeConfigIndexedValuePrefix(attributeKey); prefix != "" {
+			attributeValue = prefix + indexString
+		}
+		if oid := configHelper.getAttributeConfigOID(attributeKey); oid != "" {
+			indexedAttrKey := indexedAttributeKey{
+				parentOID: oid,
+				oidIndex:  indexString,
+			}
+			attributeValue = indexedAttributeValues[indexedAttrKey]
+		}
+		if attributeValue == "" {
+			return nil, errors.New(errMsgAttributeEmptyValue)
+		}
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
```

Comment:
What's this case handling? Above, we retrieved the attribute from config helper and set it to `attribute.Name`, and then here we overwrite it.

Reply:
This is especially confusing because we actually use the first key to access the config helpers in the conditions above.

Reply:
Ok, make sense. A comment would be helpful on this because it's not obvious.